pkg/controllers/management/clusterdeploy/clusterdeploy.go

```diff
@@ -422,8 +473,19 @@ func (cd *clusterDeploy) deployAgent(cluster *apimgmtv3.Cluster) error {
 
 	logrus.Tracef("clusterDeploy: deployAgent: detected a change in desired cluster agent manifest")
 
+	// pcChanged and pcCreated indicate that 'managePriorityClass' function has recently recreated
+	// the downstream PC, so there is no need to check a second time.
+	pcExists := pcChanged || pcCreated
+	if !pcExists {
```

Comment:
Would we need to proceed if `pcDeleted` is true? I am concerned we might redeploy unnecessarily in that case

Reply:
If the pc is deleted then we will need to redeploy the agent to remove the reference from the deployment. `ensurePriorityClass` will just exit early and return false if it doesn't find a priority class in the spec after calling `AgentSchedulingCustomizationEnabled`.